syft/internal/packagemetadata/names.go

```diff
@@ -109,6 +109,7 @@ var jsonTypes = makeJSONTypes(
 	jsonNamesWithoutLookup(pkg.RustBinaryAuditEntry{}, "rust-cargo-audit-entry", "RustCargoPackageMetadata"), // the legacy value is split into two types, where the other is preferred
 	jsonNames(pkg.WordpressPluginEntry{}, "wordpress-plugin-entry", "WordpressMetadata"),
 	jsonNames(pkg.LuaRocksPackage{}, "luarocks-package"),
+	jsonNames(pkg.TerraformLockEntry{}, "terraform-lock-entry"),
```

Comment:
re: https://github.com/anchore/syft/pull/3378/files#r1854294491

this would also be `terraform-lock-provider-entry`